src/main/java/com/coveo/spillway/trigger/AbstractLimitTrigger.java

```diff
@@ -56,13 +60,26 @@ public AbstractLimitTrigger(LimitTriggerCallback callback) {
    * @return True if the limit is triggered, false otherwise
    */
   protected abstract <T> boolean triggered(
-      T context, int cost, int currentLimitValue, LimitDefinition limitDefinition);
+      T context, int currentLimitValue, LimitDefinition limitDefinition);
 
   @Override
   public <T> void callbackIfRequired(
-      T context, int cost, int currentLimitValue, LimitDefinition limitDefinition) {
-    if (triggered(context, cost, currentLimitValue, limitDefinition)) {
+      T context,
+      int cost,
+      Instant timestamp,
+      int currentLimitValue,
+      LimitDefinition limitDefinition) {
+    Instant currentBucket = calculateBucket(timestamp, limitDefinition.getExpiration());
+
+    if (triggered(context, currentLimitValue, limitDefinition)
```

Comment:
Not sure its very intuitive, plus its more code to verify that the increase is valid...
Plus another valid example would be a async synchronisation, for example a limit of 10. The local value is 8, it sync and then become 12. If you try call it, it wont trigger the callback since 12 - 1 > 10...